app/renderer/reducers/contextMenuReducer.js

```diff
@@ -33,6 +41,45 @@ const validateState = function (state) {
   return state
 }
 
+function generateMuteFrameList (framePropsList, muted) {
+  return framePropsList.map((frameProp) => {
+    return {
+      frameKey: frameProp.get('key'),
+      tabId: frameProp.get('tabId'),
+      muted: muted && frameProp.get('audioPlaybackActive') && !frameProp.get('audioMuted')
+    }
+  })
+}
+
+const onTabPageMenu = function (state, action) {
+  action = validateAction(action)
+  state = validateState(state)
+
+  const frames = frameStateUtil.getNonPinnedFrames(state) || Immutable.List()
+  const tabsPerPage = Number(getSetting(settings.TABS_PER_PAGE))
+  const tabPageFrames = frames.slice(action.get('index') * tabsPerPage, (action.get('index') * tabsPerPage) + tabsPerPage)
+
+  const template = [{
+    label: locale.translation('unmuteTabs'),
+    click: () => {
+      windowActions.muteAllAudio(generateMuteFrameList(tabPageFrames, false))
+    }
+  }, {
+    label: locale.translation('muteTabs'),
+    click: () => {
+      windowActions.muteAllAudio(generateMuteFrameList(tabPageFrames, true))
+    }
+  }, {
+    label: locale.translation('closeTabPage'),
+    click: () => {
+      windowActions.closeFrames(tabPageFrames)
+    }
+  }]
+
+  const tabPageMenu = Menu.buildFromTemplate(template)
+  tabPageMenu.popup(getCurrentWindow())
```

Comment:
Would be great to have a unit test that ensures when `windowConstants.WINDOW_ON_TAB_PAGE_MENU` happens that `electron.remote.Menu.buildFromTemplate` fires and that `popup` is called on the instance that it returns 😄